plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -626,4 +637,53 @@ static Tuple<Long, Long> numberOfMultiparts(final long totalSize, final long par
             return Tuple.tuple(parts + 1, remaining);
         }
     }
+
+    @Override
+    public void softDeleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOException {
+        softDeleteBlobsIgnoringIfNotExists(blobNames, true);
+    }
+
+    public boolean softDeleteEnabled() {
+        return blobStore.softDeleteEnabled();
+    }
+
+    public void softDeleteBlobsIgnoringIfNotExists(List<String> blobNames, boolean relative) throws IOException {
+        if (softDeleteEnabled() == false) {
+            deleteBlobsIgnoringIfNotExists(blobNames);
+            return;
+        }
+
+        final Set<String> outstanding;
+        if (blobNames.isEmpty()) {
+            return;
+        }
+        if (relative) {
+            outstanding = blobNames.stream().map(this::buildKey).collect(Collectors.toSet());
+        } else {
+            outstanding = new HashSet<>(blobNames);
+        }
+
+        try (AmazonS3Reference clientReference = blobStore.clientReference()) {
+            for (String key : outstanding) {
+                SocketAccess.doPrivilegedVoid(() -> {
+                    PutObjectTaggingRequest putObjectTaggingRequest = PutObjectTaggingRequest.builder()
+                        .bucket(blobStore.bucket())
+                        .key(key)
+                        .tagging(blobStore.softDeleteTagging())
+                        .build();
+                    clientReference.get().putObjectTagging(putObjectTaggingRequest);
+                });
+            }
```

Comment:
Could this rather be exposed as a feature to add tags? For remote store, we can access this method for adding tags for deletion. Since there is an additional step of creation a rule or a policy, it is not an soft deletion feature offered by the remote store, but us using some other remote store features to implement soft deletion.